it/uk/gov/hmrc/personaldetailsvalidation/RepoControlServiceISpec.scala

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmrc.personaldetailsvalidation
+
+import akka.Done
+import cats.data.EitherT
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatestplus.play.guice.GuiceOneServerPerSuite
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.{HeaderCarrier, SessionId}
+import uk.gov.hmrc.personaldetailsvalidation.model.{PersonalDetailsWithNino, SuccessfulPersonalDetailsValidation, ValidationId}
+import uk.gov.hmrc.personaldetailsvalidation.services.{AssociationService, Encryption, PersonalDetailsValidatorService, RepoControlService}
+
+import java.time.format.DateTimeFormatter
+import java.time.{LocalDate, LocalDateTime}
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.crypto.PlainText
+
+class RepoControlServiceISpec extends AnyWordSpec
+  with Matchers
+  with BeforeAndAfterEach
+  with GuiceOneServerPerSuite
+  with ScalaFutures {
+
+  val config: PersonalDetailsValidationMongoRepositoryConfig = app.injector.instanceOf[PersonalDetailsValidationMongoRepositoryConfig]
+  val ec: ExecutionContext = scala.concurrent.ExecutionContext.global
+  lazy val associationRepository: AssociationMongoRepository = app.injector.instanceOf[AssociationMongoRepository]
+  lazy val pdvRepository: PdvRepository = app.injector.instanceOf[PdvRepository]
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+  }
+
+  "RepoControlService" should {
+    "save an instant of Association and personal details" in new Setup {
+
+      val result: EitherT[Future, Exception, Done] = repoControlService.insertPDVAndAssociationRecord(personalDetailsValidation,Some(testCredId),headerCarrier)(ec,encryption)
+
+      result.value.futureValue
+
+      associationService.getRecord(encryptedCredId, encryptedSessionID).futureValue match {
+        case Some(retrieved) =>
+          retrieved.credentialId shouldBe encryptedCredId
+          retrieved.sessionId shouldBe encryptedSessionID
+          retrieved.validationId shouldBe testValidationId.toString
```

Comment:
can we validate the localdatetime inserted, I did something similar in FPR DB tests. syntax is something like 
val getCurrentDateime = localdatetime.now
do test
get from db
val getCurrentDateTimeAfter = localdatetime.now
//test to make sure record from db is after or equal getCurrentDateime
// test to make sure record from db is equal to or before getCurrentDateTimeAfter

Reply:
local date time is now tested as well